### PR TITLE
wandb/tensorboard loggers set default init to False

### LINKED ISSUE #934: Several wandb init
**Describe the bug**

Hi thanks for the lib! When trying the script https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_kv_cache/llama3_fp8_kv_example.py, I see something weird: Several wandb init happens. IMHO it seems to be only a single run, so at most one init looks reasonable.

```
2024-11-26T05:25:38.849137+0000 | _check_create_state | INFO - State created for compression lifecycle
ERROR:wandb.jupyter:Failed to detect the name of this notebook, you can set it manually with the WANDB_NOTEBOOK_NAME environment variable to enable code saving.
wandb: Currently logged in as: ch271828n. Use `wandb login --relogin` to force relogin
Waiting for wandb.init()...
wandb version 0.18.7 is available! To upgrade, please run: $ pip install wandb --upgrade
Tracking run with wandb version 0.17.5
Run data is saved locally in /host_home/research/code/research_mono/notebooks/math_ai/ad_hoc/wandb/run-20241126_052540-ggjblhrs
Syncing run [glorious-mountain-19](https://wandb.ai/ch271828n/uncategorized/runs/ggjblhrs) to [Weights & Biases](https://wandb.ai/ch271828n/uncategorized) ([docs](https://wandb.me/run))
View project at https://wandb.ai/ch271828n/uncategorized
View run at https://wandb.ai/ch271828n/uncategorized/runs/ggjblhrs
2024-11-26T05:25:52.860459+0000 | pre_initialize_structure | INFO - Compression lifecycle structure pre-initialized for 0 modifiers
Finishing last run (ID:ggjblhrs) before initializing another...
View run glorious-mountain-19 at: https://wandb.ai/ch271828n/uncategorized/runs/ggjblhrs
View project at: https://wandb.ai/ch271828n/uncategorized
Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
Find logs at: ./wandb/run-20241126_052540-ggjblhrs/logs
The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
Successfully finished last run (ID:ggjblhrs). Initializing new run:
Waiting for wandb.init()...
wandb version 0.18.7 is available! To upgrade, please run: $ pip install wandb --upgrade
Tracking run with wandb version 0.17.5
Run data is saved locally in /host_home/research/code/research_mono/notebooks/math_ai/ad_hoc/wandb/run-20241126_052552-um22ijva
Syncing run [vague-universe-20](https://wandb.ai/ch271828n/uncategorized/runs/um22ijva) to [Weights & Biases](https://wandb.ai/ch271828n/uncategorized) ([docs](https://wandb.me/run))
View project at https://wandb.ai/ch271828n/uncategorized
View run at https://wandb.ai/ch271828n/uncategorized/runs/um22ijva
2024-11-26T05:26:12.416123+0000 | pre_initialize_structure | INFO - Compression lifecycle structure pre-initialized for 0 modifiers
Finishing last run (ID:um22ijva) before initializing another...
View run vague-universe-20 at: https://wandb.ai/ch271828n/uncategorized/runs/um22ijva
View project at: https://wandb.ai/ch271828n/uncategorized
Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
Find logs at: ./wandb/run-20241126_052552-um22ijva/logs
The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
Successfully finished last run (ID:um22ijva). Initializing new run:
Waiting for wandb.init()...
wandb version 0.18.7 is available! To upgrade, please run: $ pip install wandb --upgrade
Tracking run with wandb version 0.17.5
Run data is saved locally in /host_home/research/code/research_mono/notebooks/math_ai/ad_hoc/wandb/run-20241126_052612-l8wousmz
Syncing run [fast-water-21](https://wandb.ai/ch271828n/uncategorized/runs/l8wousmz) to [Weights & Biases](https://wandb.ai/ch271828n/uncategorized) ([docs](https://wandb.me/run))
View project at https://wandb.ai/ch271828n/uncategorized
View run at https://wandb.ai/ch271828n/uncategorized/runs/l8wousmz
[2024-11-26 05:26:33,277] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[/opt/conda/lib/python3.11/site-packages/llmcompressor/transformers/finetune/session_mixin.py:95](https://192.168.0.109:8888/opt/conda/lib/python3.11/site-packages/llmcompressor/transformers/finetune/session_mixin.py#line=94): FutureWarning: `tokenizer` is deprecated and will be removed in version 5.0.0 for `Trainer.__init__`. Use `processing_class` instead.
  super().__init__(**kwargs)
 [WARNING]  Please specify the CUTLASS repo directory as environment variable $CUTLASS_PATH
 [WARNING]  sparse_attn requires a torch version >= 1.5 and < 2.0 but detected 2.5
 [WARNING]  using untested triton version (3.1.0), only 1.0.0 is known to be compatible
[/opt/conda/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py:47](https://192.168.0.109:8888/opt/conda/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py#line=46): FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @autocast_custom_fwd
[/opt/conda/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py:66](https://192.168.0.109:8888/opt/conda/lib/python3.11/site-packages/deepspeed/runtime/zero/linear.py#line=65): FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @autocast_custom_bwd
2024-11-26T05:26:34.104132+0000 | one_shot | INFO - *** One Shot ***
Finishing last run (ID:l8wousmz) before initializing another...
View run fast-water-21 at: https://wandb.ai/ch271828n/uncategorized/runs/l8wousmz
View project at: https://wandb.ai/ch271828n/uncategorized
Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
Find logs at: ./wandb/run-20241126_052612-l8wousmz/logs
The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
Successfully finished last run (ID:l8wousmz). Initializing new run:
Waiting for wandb.init()...
wandb version 0.18.7 is available! To upgrade, please run: $ pip install wandb --upgrade
Tracking run with wandb version 0.17.5
Run data is saved locally in /host_home/research/code/research_mono/notebooks/math_ai/ad_hoc/wandb/run-20241126_052634-6zgyfa3j
Syncing run [desert-firefly-22](https://wandb.ai/ch271828n/uncategorized/runs/6zgyfa3j) to [Weights & Biases](https://wandb.ai/ch271828n/uncategorized) ([docs](https://wandb.me/run))
View project at https://wandb.ai/ch271828n/uncategorized
View run at https://wandb.ai/ch271828n/uncategorized/runs/6zgyfa3j
2024-11-26T05:26:53.409986+0000 | _check_compile_recipe | INFO - Recipe compiled and 1 modifiers created
2024-11-26T05:26:53.535971+0000 | _calibrate | INFO - Running QuantizationModifier calibration with 10 samples...
100%|██████████| 10[/10](https://192.168.0.109:8888/10) [00:05<00:00,  1.86it[/s](https://192.168.0.109:8888/s)]
manager stage: Modifiers initialized
2024-11-26T05:26:58.922362+0000 | initialize | INFO - Compression lifecycle initialized for 1 modifiers
manager stage: Modifiers finalized
2024-11-26T05:26:58.923158+0000 | finalize | INFO - Compression lifecycle finalized for 1 modifiers
```

**Expected behavior**

**Environment**
Include all relevant environment information:
1. OS [e.g. Ubuntu 20.04]:
2. Python version [e.g. 3.7]:
3. LLM Compressor version or commit hash [e.g. 0.1.0, `f7245c8`]:
4. ML framework version(s) [e.g. torch 2.3.1]:
5. Other Python package versions [e.g. vLLM, compressed-tensors, numpy, ONNX]:
6. Other relevant environment information [e.g. hardware, CUDA version]:

**To Reproduce**
Exact steps to reproduce the behavior:

**Errors**
If applicable, add a full print-out of any errors or exceptions that are raised or include screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here. Also include any relevant files.

### LINKED ISSUE #976: Wandb logging cannot be disabled
**Describe the bug**
It is not possible to disable logging through wandb and tensorflow. While tensorflow is silent, wandb requires interaction, so it is impossible to automate the script due to this problem. The only workaround I found is using `WANDB_DISABLED=true` when running the script.

**Expected behavior**
Since llm-compressor uses `HfArgumentParser` under hood, one would expect that passing `report_to="none"` to the `oneshot(...)` call would disable all logging. However, the problem is there, and when I run the script (more in the "To Reproduce" section), I get the following prompt from the console:
```
wandb: (1) Create a W&B account
wandb: (2) Use an existing W&B account
wandb: (3) Don't visualize my results
wandb: Enter your choice: 
```

**Environment**
- Ubuntu 22.04.3
- Python 3.10
- List of all packages:
```
Package                           Version
--------------------------------- -------------
accelerate                        1.2.0
annotated-types                   0.7.0
clearml                           1.16.5
cloudpickle                       3.1.0
compressed-tensors                0.8.0
datasets                          3.1.0
diskcache                         5.6.3
fastapi                           0.115.6
gguf                              0.10.0
h11                               0.14.0
httpcore                          1.0.7
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   0.26.5
importlib_metadata                8.5.0
interegular                       0.3.3
jiter                             0.8.2
lark                              1.2.2
llmcompressor                     0.3.0
lm-format-enforcer                0.10.9
mistral_common                    1.5.1
msgspec                           0.18.6
numpy                             1.26.4
nvidia-cublas-cu12                12.4.5.8
nvidia-cuda-cupti-cu12            12.4.127
nvidia-cuda-nvrtc-cu12            12.4.127
nvidia-cuda-runtime-cu12          12.4.127
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu12                 11.2.1.3
nvidia-curand-cu12                10.3.5.147
nvidia-cusolver-cu12              11.6.1.9
nvidia-cusparse-cu12              12.3.1.170
nvidia-ml-py                      12.560.30
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu12                  12.4.127
openai                            1.57.3
opencv-python-headless            4.10.0.84
outlines                          0.0.46
partial-json-parser               0.2.1.1.post4
pillow                            10.4.0
prometheus-fastapi-instrumentator 7.0.0
py-cpuinfo                        9.0.0
pyairports                        2.1.1
pycountry                         24.6.1
pydantic                          2.10.3
pydantic_core                     2.27.1
pynvml                            11.5.3
python-dotenv                     1.0.1
ray                               2.40.0
requests                          2.32.3
safetensors                       0.4.5
sentencepiece                     0.2.0
starlette                         0.41.3
sympy                             1.13.1
tiktoken                          0.7.0
tokenizers                        0.20.3
torch                             2.5.1
torchvision                       0.20.1
tqdm                              4.67.0
tqdm-multiprocess                 0.0.11
transformers                      4.46.3
triton                            3.1.0
typing_extensions                 4.12.2
uvicorn                           0.32.1
uvloop                            0.21.0
vllm                              0.6.4.post1
watchfiles                        1.0.3
websockets                        14.1
xformers                          0.0.28.post3
zipp                              3.21.0
```

**To Reproduce**
To reproduce the issue, use the following script, which is a modified version of the one in the README.md
```
from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.modifiers.smoothquant import SmoothQuantModifier
from llmcompressor.transformers import oneshot
from transformers import AutoModelForCausalLM

recipe = [
    SmoothQuantModifier(smoothing_strength=0.8),
    GPTQModifier(scheme="W8A8", targets="Linear", ignore=["lm_head"]),
]

oneshot(
    model="TinyLlama/TinyLlama-1.1B-Chat-v1.0",
    dataset="open_platypus",
    recipe=recipe,
    output_dir="TinyLlama-1.1B-Chat-v1.0-INT8",
    max_seq_length=2048,
    num_calibration_samples=512,
    report_to="none"
)
```

### PR DESCRIPTION
**Pull request by brian-dellabetta**
SUMMARY:
If `wandb` is installed, an empty logging config will default to including wandb, causing workflows to hang on the following prompt:
```
wandb: Using wandb-core as the SDK backend.  Please refer to https://wandb.me/wandb-core for more information.
wandb: (1) Create a W&B account
wandb: (2) Use an existing W&B account
wandb: (3) Don't visualize my results
wandb: Enter your choice: 
```
This resolves by setting default behavior to False, expecting user to provide a config in order to use `wandb`.
Resolves #976 
Resolves #934 

TEST PLAN:
No new src code

### CODE REVIEW

**Review by kylesayrs — APPROVED** [review]

**Review by horheynm — APPROVED** [review]

**Comment by markurtz** [comment]
@brian-dellabetta do we not have a better pathway for determining if wandb is configured and in a non-blocking usable state? Since the user doesn't have easy pathways into configuring the loggers currently, especially for our oneshot APIs, I'm a bit cautious of essentially removing metric logging for those completely.

Separately, though, we do need to refactor how we're handling loggers and their general setup, so that can wait til then.

**Comment by brian-dellabetta** [comment]
> @brian-dellabetta do we not have a better pathway for determining if wandb is configured and in a non-blocking usable state? Since the user doesn't have easy pathways into configuring the loggers currently, especially for our oneshot APIs, I'm a bit cautious of essentially removing metric logging for those completely.
> 
> Separately, though, we do need to refactor how we're handling loggers and their general setup, so that can wait til then.

@markurtz this part of the codebase is super opaque to me, and because `oneshot` just takes `kwargs` it is not clear to me where a user could configure this. Right now this is blocking two users from running anything, so this is a hotfix that should be considered more deeply later on (probably overlapping current work of Kyle and George for their refactoring). 

But i think this indicates there is a whole lot of code involved in training that axolotl/llm-foundry/instructlab take care of, and that we're better off connecting into them for more sophisticated workflows that take care of all that
